### PR TITLE
[feat] Add support to skip arguments

### PR DESCRIPTION
**Pull request by inishchith**
Add optional parameter to skip arguments

example:

```python3
@dd.record_function(frequency=0.5)
def api1(a: float = 1.0, b: float = 2.0) -> float:
    return a + b

@dd.record_function(frequency=0.5, skip_args=["b"])
def api1(a: float = 1.0, b: float = 2.0) -> float:
    return a + b
```

### CODE REVIEW

File: src/dataset_dog/api.py [hunk 1]
```diff
@@ -26,7 +25,7 @@ def _submit_callback(
         function_name: str,
         args: Tuple[Any, ...],
         kwargs: Dict[str, Any],
-        res: Any,
+        res: Any
```
Comment (vidursatija) [inline src/dataset_dog/api.py]: please use `./pre-commit.sh` to fix linting issues

File: src/dataset_dog/api.py [hunk 2]
```diff
@@ -4,13 +4,12 @@
 import pickle
 import random
 import base64
-from typing import Any, Callable, Dict, Tuple
+from typing import Any, Callable, Dict, Tuple, List
```
Comment (vidursatija) [inline src/dataset_dog/api.py]: typing isn't sorted. please use the pre-commit file

File: src/dataset_dog/api.py [hunk 3]
```diff
@@ -39,7 +38,22 @@ def _submit_callback(
         )
         self.worker.submit(function_info)
 
-    def record_function(self, frequency: float):
+    @staticmethod
+    def filter_arguments(args: Tuple[str], kwargs: Dict[str, Any], skip_args: List[str]) -> Tuple[List[str], Dict[str, Any]]:
```
Comment (vidursatija) [inline src/dataset_dog/api.py]: args aren't a string tuple. They can be anything. You'll have to use the inspect module(https://stackoverflow.com/questions/218616/how-to-get-method-parameter-names) to get the args name and then do the filtering

File: README.md [hunk 4]
```diff
@@ -7,7 +7,7 @@ python3 -m pip install git+https://github.com/vidursatija/dataset-dog-python.git
 ```
 
 ## Usage
-```python
+```python3
```
Comment (vidursatija) [inline README.md:10]: this doesn't work

File: README.md [hunk 5]
```diff
@@ -18,6 +18,14 @@ def api1(a: float = 1.0, b: float = 2.0) -> float:
     return a + b
 ```
 
+- Skip recording sensitive arguments
+
+```python3
```
Comment (vidursatija) [inline README.md:23]: python3 isn't valid

**Comment by vidursatija** [comment]
Please merge master again because I've pushed some changes

File: README.md (hunk 4 above)
Comment (inishchith) [reply README.md:10]: it does work - try edit on the PR description, it has python3

File: README.md (hunk 5 above)
Comment (inishchith) [reply README.md:23]: x-post
> it does work - try edit on the PR description, it has python3

File: README.md [hunk 6]
```diff
@@ -17,3 +17,20 @@ dd = DatasetDog("<API SERVER>", "<PROJECT ID>", "<API SECRET>")
 def api1(a: float = 1.0, b: float = 2.0) -> float:
     return a + b
 ```
+<<<<<<< HEAD
```
Comment (vidursatija) [inline README.md]: conflict

File: README.md [hunk 7]
```diff
@@ -17,3 +17,20 @@ dd = DatasetDog("<API SERVER>", "<PROJECT ID>", "<API SECRET>")
...
+def api1(a: float = 1.0, b: float = 2.0) -> float:
+    return a + b
+```
+
+## TODO
+- Documentation
+- Examples
+- Pypi
+=======
+>>>>>>> master
```
Comment (vidursatija) [inline README.md]: conflict

File: src/dataset_dog/api.py [hunk 8]
```diff
@@ -45,7 +45,38 @@ def _submit_callback(
             return
         self.worker.submit(function_info)
 
-    def record_function(self, frequency: float):
+    @staticmethod
+    def get_filtered_arguments(
+        function,
+        args,
+        kwargs,
```
Comment (vidursatija) [inline src/dataset_dog/api.py]: please add typing here

**Review by vidursatija — CHANGES_REQUESTED** [review]

**Review by vidursatija — APPROVED** [review]

**Pull request merged**